source/simulator/core.cc

```diff
@@ -1976,6 +1976,26 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          // Identical to IMPES except does not solve Stokes equation
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+       
+          std::string fileName = parameters.input_directory + "/" + "solution-" + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+          triangulation.load(fileName.c_str());
+          parallel::distributed::SolutionTransfer<dim, LinearAlgebra::BlockVector> sol_trans(dof_handler);
+          sol_trans.deserialize (solution);
+          //triangulation.clear();
```

Comment:
>  Is your binary-input for Stokes only, Stokes+Temperature+Composition, or either? It probably makes the most sense to read in all variables, right? If yes, "prescribed Stokes" is obviously the wrong place.

We are interested in binary-input for Stokes+Temperature data at each time step (in the future we may be interested in compositional field as well). We are considering reading in all the variables. Where is the right place to read in binary data?